### PR TITLE
feat(snap): add git as craft.git to rockcraft

### PR DESCRIPTION
**Pull request by dariuszd21**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

This PR adds stages `git` in the `rockcraft` snap package, to enable `git`-based operations for `craft` libraries.

(CRAFT-3548)

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -53,6 +53,26 @@ parts:
...
+    plugin: nil
+    stage-packages: [git]
+    build-attributes:
+      - enable-patchelf
+    prime:
+      - "usr/lib/git-core"
+      - "usr/share/git-core"
+      - "usr/lib/x86_64-linux-gnu"
+      # perl is part of the core22 / core24
+      - "-usr/lib/x86_64-linux-gnu/perl"
```
Comment (tigarmo) [inline snap/snapcraft.yaml:72]: I suspect these removals aren't needed; if you specify any inclusions when only those paths are primed

File: tests/spread/rockcraft/craftgit/task.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+summary: craft.git tool test
+
+execute: |
+  run_rockcraft pack
+
+  test -f craftgit-test_latest_amd64.rock
+  test ! -d parts -a ! -d stage -a ! -d prime
```
Comment (tigarmo) [inline tests/spread/rockcraft/craftgit/task.yaml:7]: this line seems unrelated to the test itself

File: tests/spread/rockcraft/craftgit/rockcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: craftgit-test
```
Comment (tigarmo) [inline tests/spread/rockcraft/craftgit/rockcraft.yaml:1]: 👍🏾

**Review by tigarmo — APPROVED** [review]

File: snap/snapcraft.yaml (hunk 1 above)
Comment (dariuszd21) [reply snap/snapcraft.yaml:72]: I'll try to include only libcurl and see if that will work, then I could remove the exclusions entirely:
```
- "usr/lib/x86_64-linux-gnu/libcurl-gnutls*"
```

Comment (dariuszd21) [reply snap/snapcraft.yaml:72]: In the end I needed to include the following to get rid of exclusion:
```yaml
      - "usr/lib/x86_64-linux-gnu/libcurl-gnutls*"
      - "usr/lib/x86_64-linux-gnu/libpsl*"
      - "usr/lib/x86_64-linux-gnu/librtmp*"
      - "usr/lib/x86_64-linux-gnu/libnghttp2*"
```

File: tests/spread/rockcraft/craftgit/task.yaml (hunk 2 above)
Comment (dariuszd21) [reply tests/spread/rockcraft/craftgit/task.yaml]: Removed that, and added a restore that deletes a created rock instead

**Review by mr-cal — APPROVED** [review]
